MapboxNavigation/CarPlayManager.swift

```diff
@@ -509,14 +511,19 @@ extension CarPlayManager: CPMapTemplateDelegate {
         var service: NavigationService
         
         let desiredSimulationMode: SimulationMode = simulatesLocations ? .always : .onPoorGPS
-        if let override = delegate?.carPlayManager(self, navigationServiceAlong: route, desiredSimulationMode: desiredSimulationMode) {
-            service = override
+        if let navService = navigationService {
+            service = navService
         } else {
-            service = MapboxNavigationService(route: route, simulating: desiredSimulationMode)
+            if let override = delegate?.carPlayManager(self, navigationServiceAlong: route, desiredSimulationMode: desiredSimulationMode) {
```

Comment:
This can be written more succinctly as:

```swift
if let navigationService = navigationService ?? delegate?.carPlayManager(self, navigationServiceAlong: route, desiredSimulationMode: desiredSimulationMode) {
    self.navigationService = navigationService
} else {
    navigationService = MapboxNavigationService(route: route, simulating: desiredSimulationMode)
}
```

or even:

```swift
self.navigationService = navigationService ??
    delegate?.carPlayManager(self, navigationServiceAlong: route, desiredSimulationMode: desiredSimulationMode) ??
    MapboxNavigationService(route: route, simulating: desiredSimulationMode)
```